core/templates/base-components/oppia-footer.component.ts

```diff
@@ -53,6 +63,40 @@ export class OppiaFooterComponent {
       return 'https://medium.com/oppia-org';
     }
   }
+
+  validateEmailAddress(): boolean {
+    let regex = new RegExp(AppConstants.EMAIL_REGEX);
+    return regex.test(String(this.emailAddress));
+  }
+
+  subscribeToMailingList(): void {
+    this.mailingListBackendApiService.subscribeUserToMailingList(
+      String(this.emailAddress),
+      // Most of the times, since the name is not provided
+      // so it is null. In such case placeholder name is set to UserX.
+      String(this.name) ?? 'UserX',
```

Comment:
This seems bad. Don't pass fake data to the API. Just send a null name.

Reply:
I was also not sure about this approach, thanks for pointing it out. :)